### DISCUSSION TITLE
Help me understanding configuration please

### DISCUSSION THREAD

**Discussion by andyy-mishell** [quartznet/quartznet#1527] [Q&A]
Greetings everyone, 

So let me explain this staright

For my job, I was assigned to investigate and document this library (in the easiest and most summarized way) in order to make a guidebook, example applications and train my team. So far I have covered general concepts that I found checking the features and frequently asked questions sections from the webpage, now I'm redacting the quick guide part and I'm kinda stuck on the configuration section:

- There is 'Fluent scheduler builder API' part that I don't know how to track down, it mention something about C# fluent API ¿What is that? or the **NameValueCollection** parameter in the scheduler factory ¿What is this factory? ¿Where is it? Also there is a code snipset below ¿What is this code and for what's for? ¿Is this code somewhere on the app files or the dll files? ¿Should I copy/paste this code somewhere else? 
- Also, It's mentioned this **quartz.config** file for basic configuration but I can't find it within my project directory

So basically, I can't get mostly all of this section and I know maybe this could be a dumb question (I'm a noob) but I would be gratefull to receive your help and understanding because I can't go forward in my task. 

Thank you.

<img width="958" alt="1" src="https://user-images.githubusercontent.com/70854221/155385419-e20b5117-db02-4f95-ae4a-7499ae273632.png">

**Comment by lahma** [comment]
> * There is 'Fluent scheduler builder API' part that I don't know how to track down, it mention something about C# fluent API ¿What is that?

Well it's less advertised builder, [source code here](https://github.com/quartznet/quartznet/blob/main/src/Quartz/SchedulerBuilder.cs). It basically allows you to build the scheduler with more strongly typed API without need to pass the property names as string using the `NameValueCollection` constructor approach. This is what the DI configuration story uses under the hood mostly.

>  or the **NameValueCollection** parameter in the scheduler factory ¿What is this factory? ¿Where is it? Also there is a code snipset below ¿What is this code and for what's for? ¿Is this code somewhere on the app files or the dll files? ¿Should I copy/paste this code somewhere else?

Maybe you should read also [the tutorial](https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/using-quartz.html), quick start is quick start which also expects you to dig in more by browsing the API documentation etc. 

> * Also, It's mentioned this **quartz.config** file for basic configuration but I can't find it within my project directory

You should also definitely open the Quartz sources in your favorite development environment and explore the examples. They are the projects starting with `Quartz.Examples`, part of [the src directory](https://github.com/quartznet/quartznet/tree/main/src).
